mobile/src/main/java/ch/epfl/sdp/mobile/ui/profile/ProfileScreen.kt

```diff
@@ -114,7 +114,11 @@ fun <C : ChessMatch> ProfilePicture(
  * @param modifier the [Modifier] for this composable.
  */
 @Composable
-fun UnfollowButton(onClick: () -> Unit, modifier: Modifier = Modifier) {
+fun <C : ChessMatch> UnfollowButton(
+    state: VisitedProfileScreenState<C>,

```

Comment:
You should not pass the whole state to the composable. What would you think of passing it a text: String parameter instead, and let the caller decide what to display based on the state ?